src/pint/toa.py

```diff
@@ -2584,3 +2581,250 @@ def merge_TOAs(TOAs_list, strict=False):
     nt.merged = True
 
     return nt
+
+
+def get_TOAs_array(
+    times,
+    obs,
+    scale=None,
+    errors=1 * u.us,
+    freqs=np.inf * u.MHz,
+    flags=None,
+    model=None,
+    ephem=None,
+    include_bipm=True,
+    bipm_version=bipm_default,
+    include_gps=True,
+    planets=False,
+    tdb_method="default",
+    commands=None,
+    hashes=None,
+    limits="warn",
+    **kwargs,
+):
+    """Load and prepare TOAs for PINT use from an array of times.
+
+    Creates TOAs from a an array of times, applies clock corrections, computes
+    key values (like TDB), computes the observatory position and velocity
+    vectors.
```

Comment:
Should we say that this requires that all of the TOAs be at the same observatory? Also, does this work properly for an observatory whose position is a function of time (e.g. a spacecraft)?

Reply:
The latest docstring does say that:
```
    Note that this requires all TOAs to be measured at a single observatory.  If you
    wish to combine multiple data-sets use :func:`pint.toa.merge_TOAs`
```
But I will test for spacecraft.

Reply:
Spacecraft work fine. I put in a test and added it to the docs.

Reply:
Actually, since this works I simplified the Fermi workflow:
```
t, obs, energies, weights = load_Fermi_Times(eventfile, weightcolumn="PSRJ0030+0451")
toas = toa.get_TOAs_array(t, obs, include_gps=False,
                include_bipm=False, planets=False, ephem="DE405",
                flags=[{"energy": str(e), "weight": str(w)} for e, w in zip(energies.to_value(u.MeV), weights)],
                )
```
It's not hugely different but could be slightly more efficient.  If you think this is useful I can replicate this for the other functions in `event_toas.py` (should the Fermi functions maybe be in there?)